parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetReader.java

```diff
@@ -133,6 +149,97 @@ public T read() throws IOException {
     }
   }
 
+  private boolean readVectors(ColumnVector[] vectors, MessageType[] columns) throws IOException {
+    if (reader != null && reader.nextBatch(vectors, columns)) {
+      return true;
+    } else {
+      initReader();
+      if (reader == null) {
+        return false;
+      } else {
+        return readVectors(vectors, columns);
+      }
+    }
+  }
+
+  private boolean readVector(ObjectColumnVector<T> vector) throws IOException {
+    if (reader != null && reader.nextBatch(vector)) {
+      return true;
+    } else {
+      initReader();
+      if (reader == null) {
+        return false;
+      } else {
+        return readVector(vector);
+      }
+    }
+  }
+
+  /**
+   * Reads the next batch of rows. This method is used for reading complex types
+   * or arbitrary objects and calls the converters eventually to materialize the record.
+   * @param previous a row batch object to be reused by the reader if possible
+   * @param clazz the class of the record type that will be filled into the column vector
+   * @return the row batch that was read
+   * @throws java.io.IOException
+   */
+  public RowBatch nextBatch(RowBatch previous, Class<T> clazz) throws IOException {
+    RowBatch rowBatch = previous;
+    if (rowBatch == null) {
+      rowBatch = new RowBatch();
+    }
+
+    ObjectColumnVector<T> columnVector;
+    if (rowBatch.getColumns() == null) {
+      columnVector = ColumnVector.ofType(clazz);
+      rowBatch.setColumns(new ColumnVector[] { columnVector });
+    } else {
+      columnVector = (ObjectColumnVector<T>) rowBatch.getColumns()[0];
+    }
+
+    return readVector(columnVector) ? rowBatch : null;
+  }
+
+  /**
+   * Reads the next batch of rows. This method is used for reading primitive types
+   * and does not call the converters at all.
+   * @param previous a row batch object to be reused by the reader if possible
+   * @return the row batch that was read
+   * @throws java.io.IOException
+   */
+   public RowBatch nextBatch(RowBatch previous) throws IOException {
```

Comment:
we use "nextBatch" in different contexts to do different things. Let's add a verb in there.